### ISSUE TITLE
"Error setting up ML" exceptions while installing Filebeat for Windows.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/beats#23755]
**Kibana version:**
Kibana: 7.11.0 BC-5 Cloud environment

**Host OS and Browser version:**
Windows 10, All

**Preconditions:**
1.  7.11.0 BC-5 Cloud environment should be available.

**Build Details:**
```
    Artifact link used: https://staging.elastic.co/7.11.0-903dc0b6/downloads/beats/filebeat/filebeat-7.11.0-windows-x86_64.zip
    BUILD: 37827
    COMMIT: d801a7bb08e17368584d00fbd97a5d0006285b51
```
2. Download Filebeat Windows zip file and extract at C:\Program Files.

**Steps to reproduce:**
1. Open a PowerShell prompt as an Admin user and install FileBeat and run below commands:
PS > cd 'C:\Program Files\Filebeat'
PS > C:\Program Files\Filebeat> .\install-service-filebeat.ps1
2. Connect to Kibana cloud environment by providing cloud.id and cloud.auth in filebeat.yml.
3. Enable system modules from Filebeat installation directory:
PS > .\filebeat.exe modules enable system
4. Setup assets from installation directory and run:
PS > .\filebeat.exe setup -e
5. Observe "Error setting up ML" exceptions after setup command.

**Reference ticket Id:**
https://github.com/elastic/observability-test-plans/issues/88

**Expected Result:**
Filebeat should be installed as per [Filebeat quick start: installation and configuration](https://www.elastic.co/guide/en/beats/filebeat/7.x/filebeat-installation-configuration.html.)

**Screenshots:**
![4](https://user-images.githubusercontent.com/77374876/106274589-8271b200-625a-11eb-9175-d17faecb6c8c.PNG)

**Note:**
We have observed the same issue while installing Filebeat on Linux.
![3](https://user-images.githubusercontent.com/77374876/106277130-93242700-625e-11eb-961c-fc039f0e1108.JPG)

**Comment by manishgupta-qasource** [comment]
Reviewed & Assigned to @EricDavisX

**Comment by ph** [comment]
@EricDavisX I am confused by this issue, this look like a Filebeat issue why it's linked to ingest manager?

**Comment by EricDavisX** [comment]
just a habit of the test team and it got missed in their side of issue review.  i agree it looks like beats so I will remove the [tag] and label.

**Comment by EricDavisX** [comment]
I'm not familiar with this functionality though, so I don't know if it is indeed high impact or urgent for 7.11 review.  Hence I am leaving the 'high' impact and 7.11 labels for the Filebeat team to review.  @ph thanks for the help spotting it.

**Comment by urso** [comment]
@kvch Does this ring a bell?

**Comment by kvch** [comment]
No, but I would like to look into this.

**Comment by EricDavisX** [comment]
I read this closer - if it entirely blocks Filebeat from installing on all Windows OSes it would have to be a blocker for 7.11 right?  Not sure if anyone can link to what CI we have that proves Windows can work, tho it is not in this specific config

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
Today while installing Filebeat for MAC on 7.11 BC-5 Kibana cloud environment we observed few errors. 

**Build details:**
```
Artifact link used: https://staging.elastic.co/7.11.0-903dc0b6/downloads/beats/filebeat/filebeat-7.11.0-darwin-x86_64.tar.gz
Build: 37827
Commit: d801a7bb08e17368584d00fbd97a5d0006285b51
```
**Observations:** 
- "Error reading manifest file" after running- "sudo ./filebeat -e" command.

**Screenshots:**
![errr1](https://user-images.githubusercontent.com/77374876/106893953-4d5ed700-6714-11eb-99d6-c7f6a1dac2e4.JPG)

Further we tried to troubleshoot this using: "sudo chown root manifest.yml" but still getting same errors.

Please let us know if anything else is required.
Thanks
QAS

**Comment by ph** [comment]
Answer for https://github.com/elastic/beats/issues/23755#issuecomment-773308608
@amolnater-qasource your latest comment is another issue, that should have been a different issue. Now, if you look at the log message we do a strict check if the file is owner by root,  Running filebeat with `sudo` will not change the ownership you have to `chown` the files.

I don't think it's a blocker but we need to get that fixed a patch release, but I am concerned I believe the test of our modules should have raised an issue here? I am not too familiar with ML in filebeat or elasticsearch

Where `apache_ecs` is defined? I can't find any reference of that specific module in our code?

**Comment by EricDavisX** [comment]
@amolnater-qasource please do log a separate bug for the chown issue noted.  thank you.  and thank you PH.

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 
We have reported the required issue https://github.com/elastic/beats/issues/23874 .
Please let us know if anything else is required.

Thanks 
QAS

**Comment by kvch** [comment]
@ph We get such errors from time to time. I had investigated numerous issues, most of the problems were coming from incorrect permissions for the user who is setting up ML. It is not a blocker, as users can setup ML from Kibana.

**Comment by EricDavisX** [comment]
I am glad to see confirmation we have e2e tests that cover this.  If there is any documentation update we want to make regarding setup requirements and user permissions that might be valuable.  @amolnater-qasource if you can re-check and do a read through of what on-line available documentation you can, including Kibana side, please let us know if you don't find reference and where it might best fit and we can evaluate updates.

**Comment by kvch** [comment]
Here are the permissions/roles which are required for running the setup command: https://www.elastic.co/guide/en/beats/filebeat/current/privileges-to-setup-beats.html

**Comment by ph** [comment]
Thanks @kvch for clarifying the sitatation, happy to heard its a permissions and our E2E cover this :)

**Comment by amolnater-qasource** [comment]
Hi @ph 
As per feedback, we have attempted to re-install Filebeat on Windows. 
Further we have also referred [Filebeat quick start: installation and configuration](https://www.elastic.co/guide/en/beats/filebeat/7.x/filebeat-installation-configuration.html), and made few attempts to resolve this issue. However we were getting the same "ML errors".
 
So @ph it would be helpful if you could provide us syntax for setup command to grant permissions/roles provided at: [Grant privileges and roles](https://www.elastic.co/guide/en/beats/filebeat/current/privileges-to-setup-beats.html) for reducing ML errors. 

Thanks
QAS

**Comment by ph** [comment]
@amolnater-qasource Does the `filebeat_setup` user for elasticsearch has the following [privileges](https://www.elastic.co/guide/en/beats/filebeat/current/privileges-to-setup-beats.html)?

Cluster | manage_ilm | Set up and manage index lifecycle management (ILM) policy
-- | -- | --
Cluster | manage_ml | Set up Machine Learning job configurations

**Comment by amolnater-qasource** [comment]
Hi @ph

While troubleshooting the filebeat related errors, today we were able to install filebeat successfully on Windows.
Troubleshooting steps:
We used a test file in csv format and enabled type: log setting under Filebeat inputs section and provided test file at path section.
After using below commands:
```
    .\install-service-filebeat.ps1
    .\filebeat setup
    Start-Service filebeat
```
we were able to view data of csv file under kibana > Discovery Section

Hence we are closing this issue.

cc: @EricDavisX 

Thanks 
QAS
